lib/ts/station_fsm.ex

```diff
@@ -0,0 +1,343 @@
+defmodule StationFsm do
+	@moduledoc """
+	Provides implementation of the core logic of a Station.
+	"""
+
+	use Fsm, initial_state: :start, initial_data: []
+
+	# Function definitions
+
+	def process_itinerary(station_fsm, itinerary)  do
```

Comment:
1. Keep helper functions private
2. Move all the active work from events to private functions
3. Have the private functions invoked inside the code block of events